packages/react-native-web/src/exports/Image/index.js

```diff
@@ -186,33 +181,29 @@ const Image: React.AbstractComponent<
     }
   }
 
-  const [state, updateState] = React.useState(() => {
-    const uri = resolveAssetUri(source);
-    if (uri != null) {
-      const isLoaded = ImageLoader.has(uri);
-      if (isLoaded) {
-        return LOADED;
-      }
-    }
-    return IDLE;
-  });
-
+  // Only the main source is supposed to trigger onLoad/start/end events
+  // It would be ambiguous to trigger the same `onLoad` event when default source loads
+  // That's why we don't pass `onLoad` props for the fallback source hook
+  const fallbackSource = useSource({ onError }, defaultSource);
+  const mainSource = useSource(
+    { onLoad, onLoadStart, onLoadEnd, onError },
+    source
+  );
   const [layout, updateLayout] = React.useState({});
   const hasTextAncestor = React.useContext(TextAncestorContext);
   const hiddenImageRef = React.useRef(null);
   const filterRef = React.useRef(_filterId++);
-  const requestRef = React.useRef(null);
   const shouldDisplaySource =
-    state === LOADED || (state === LOADING && defaultSource == null);
+    mainSource.status === LOADED ||
+    (mainSource.status === LOADING && defaultSource == null);
   const [flatStyle, _resizeMode, filter, tintColor] = getFlatStyle(
     style,
     blurRadius,
     filterRef.current
   );
   const resizeMode = props.resizeMode || _resizeMode || 'cover';
-  const selectedSource = shouldDisplaySource ? source : defaultSource;
-  const displayImageUri = resolveAssetUri(selectedSource);
-  const imageSizeStyle = resolveAssetDimensions(selectedSource);
```

Comment:
`resolveAssetUri` was changed to `resolveSource` and it takes care or resolving the URI and the dimensions